src/RRTStar/RRTStarWrapper.cpp

```diff
@@ -0,0 +1,167 @@
+            radius = min(
+                pow(gamma / PI * log(i + 1) / (i + 1), 1 / SPACEDIM),
+                (double) rrt->step_size
+            );
+
+            // look for shortest cost path to qNew
+            for (auto node : rrt->nodes) {
+                // compute the cost for qNew through node
+                dist = rrt->distance(qNew->position, node->position);
+                cost = cost_map[node] + dist;
+
+                // only consider proximal nodes
+                if (dist > radius) {
+                    break;
+                }
+
+                // if cost less than current lowest cost, check potential link
+                if (cost < cost_map[qBest] + distBest) {
+                    if (rrt->isSegmentInObstacles(qNew->position,
+                                                  node->position)) {
+                        continue;
+                    }
+                    qBest = node;
+                    distBest = dist;
+                }
+            }
+
+            if (rrt->isSegmentInObstacles(qNew->position, qBest->position)) {
+                continue;
+            }
+            rrt->add(qBest, qNew, distBest);
+            cost_map[qNew] = cost_map[qBest] + distBest;
+
+            // rewire the tree
+            for (auto node : rrt->nodes) {
+                dist = rrt->distance(qNew->position, node->position);
+                cost = cost_map[node] + dist;
+
+                // only consider proximal nodes
+                if (dist > radius) {
+                    break;
+                }
+
+                // if cost less than current lowest cost, check potential link
+                if (cost < cost_map[node]) {
+                    if (rrt->isSegmentInObstacles(qNew->position,
+                                                  node->position)) {
+                        continue;
+                    }
+                    rrt->relink(node, qNew, dist);
+                    cost_map[node] = cost;
+                }
+            }
+            // early exit
+            if (rrt->reached()) {
+                reached = 1;
+                break;
+            }
+        }
+        Node *q;
+
+        if (reached) {
+            q = rrt->last_node;
+        }
+        else
+        {
+            // if not reached yet, then shortestPath will start from the closest
+            // node to end point.
+            q = rrt->nearest(rrt->end_pos);
+        }
+        // generate shortest path to destination.
+        while (q != NULL) {
+            rrt->path.push_back(q);
+            q = q->parent;
+        }
+
+        // return the RRT* path
+        int index = 0;
+        reverse(rrt->path.begin(), rrt->path.end());
+        for (auto node : rrt->path) {
+            xPath[index] = node->position[0];
+            yPath[index] = node->position[1];
+            index += 1;
+        }
+        xPath[index] = NAN;
+        yPath[index] = NAN;
+
+        // free the memory
+        rrt->reset();
```

Comment:
Why not just delete rrt? IMHO, there's not need for a reset method.

Reply:
Done!